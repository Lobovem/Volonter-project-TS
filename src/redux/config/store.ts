import { configureStore } from '@reduxjs/toolkit';
import contentSlice from '../slice/contentSlice';

export const store = configureStore({
  reducer: {
    content: contentSlice,
  },
});

// Выведение типов `RootState` и `AppDispatch` из хранилища
export type RootState = ReturnType<typeof store.getState>;
// Выведенные типы: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
