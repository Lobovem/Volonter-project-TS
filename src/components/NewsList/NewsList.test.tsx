import { render, screen } from '@testing-library/react';
import { NewsList } from './NewsList';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { store } from '../../redux/config/store';

describe('NewsList', () => {
  const data = [
    {
        "id": "1",
        "title": "Якась важлива новина 1",
        "desc": "За допомогою онлайн -генератора тексту ви можете обробляти свій особистий Lorem Ipsum , збагачуючи його html -елементами, які визначають його структуру, з можливістю вставляти зовнішні посилання, але не тільки. Фактично, вставляючи будь -який фантастичний текст або відомий текст, це вірш, промова, літературний уривок, текст пісні тощо, наш текстовий генератор забезпечить випадкове виділення термінів і кроків для створення власного ексклюзивного Lorem Ipsum. Фактично.",
        "photoMain": "/img/slider-news-1.jpg",
        "photoList": [
            "/img/slider-news-1.jpg",
            "/img/slider-news-2.jpg",
            "/img/slider-news-1.jpg",
            "/img/slider-news-2.jpg"
        ],
    },
    {
        "id": "2",
        "title": "Якась важлива новина 2",
        "desc": "За допомогою онлайн -генератора тексту ви можете обробляти свій особистий Lorem Ipsum , збагачуючи його html -елементами, які визначають його структуру, з можливістю вставляти зовнішні посилання, але не тільки. Фактично, вставляючи будь -який фантастичний текст або відомий текст, це вірш, промова, літературний уривок, текст пісні тощо, наш текстовий генератор забезпечить випадкове виділення термінів і кроків для створення власного ексклюзивного Lorem Ipsum. Фактично.",
        "photoMain": "/img/slider-news-2.jpg",
        "photoList": [
            "/img/slider-news-1.jpg",
            "/img/slider-news-2.jpg",
            "/img/slider-news-1.jpg",
            "/img/slider-news-2.jpg"
        ],
    }
]

  test('renders NewsList component', () => {
    const { container } = render( 
    
    <Provider store={store}>
    <Router>
    <NewsList currentItems={data}/>
    </Router>
  </Provider>)

    expect(container).toMatchSnapshot();
  });

  test('search tag h3 (decor)', () => {
    render( <Provider store={store}>
      <Router>
      <NewsList currentItems={data}/>
      </Router>
    </Provider>);
    const h3Element = screen.getByRole('heading', { level: 3 });
    expect(h3Element).toBeInTheDocument();
  });
});
