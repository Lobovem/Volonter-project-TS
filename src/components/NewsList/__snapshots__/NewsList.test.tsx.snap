// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsList renders NewsList component 1`] = `
<div>
  <div
    class="news"
  >
    <h3
      class="news__decor"
    >
      Новини
    </h3>
    <div
      class="news__wrap"
    >
      <div
        class="news__inner"
      >
        <div
          class="news__imgWrap"
        >
          <a
            href="/news/1"
          >
            <img
              alt="news.jpg"
              class="news__img"
              src="/img/slider-news-1.jpg"
            />
          </a>
        </div>
        <div
          class="news__textBox"
        >
          <div
            class="news__textWrap"
          >
            <a
              class="news__linkAlt"
              href="/news/1"
            >
              <h2
                class="titleMain"
              >
                Якась важлива новина 1
              </h2>
            </a>
            <p
              class="news__desc"
            >
              За допомогою онлайн -генератора тексту ви можете обробляти свій особистий Lorem Ipsum , збагачуючи його html -елементами, які визначають його структуру, з можливістю вставляти зовнішні посилання, але не тільки. Фактично, вставляючи будь -який фантастичний текст або відомий текст, це вірш, промова, літературний уривок, текст пісні тощо, наш текстовий генератор забезпечить випадкове виділення термінів і кроків для створення власного ексклюзивного Lorem Ipsum. Фактично.
            </p>
          </div>
          <a
            class="news__link"
            href="/news/1"
          >
            Читати повністю
          </a>
        </div>
      </div>
      <div
        class="news__inner"
      >
        <div
          class="news__imgWrap"
        >
          <a
            href="/news/2"
          >
            <img
              alt="news.jpg"
              class="news__img"
              src="/img/slider-news-2.jpg"
            />
          </a>
        </div>
        <div
          class="news__textBox"
        >
          <div
            class="news__textWrap"
          >
            <a
              class="news__linkAlt"
              href="/news/2"
            >
              <h2
                class="titleMain"
              >
                Якась важлива новина 2
              </h2>
            </a>
            <p
              class="news__desc"
            >
              За допомогою онлайн -генератора тексту ви можете обробляти свій особистий Lorem Ipsum , збагачуючи його html -елементами, які визначають його структуру, з можливістю вставляти зовнішні посилання, але не тільки. Фактично, вставляючи будь -який фантастичний текст або відомий текст, це вірш, промова, літературний уривок, текст пісні тощо, наш текстовий генератор забезпечить випадкове виділення термінів і кроків для створення власного ексклюзивного Lorem Ipsum. Фактично.
            </p>
          </div>
          <a
            class="news__link"
            href="/news/2"
          >
            Читати повністю
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
